google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java

```diff
@@ -186,9 +187,17 @@ public void apply() throws ConfigurationException {
     }
 
     try {
-      if (isRuntimeCustom() && !isValidConfigurationFile(dockerfile.getText())) {
-        throw new ConfigurationException(
-            GctBundle.getString("appengine.deployment.error.staging.dockerfile"));
+      if (isRuntimeCustom()) {
+        String dockerfileDir = dockerDirectory.getText();
+        if (dockerfileDir.isEmpty() || !Files.isDirectory(Paths.get(dockerfileDir))) {
+          throw new ConfigurationException(
+              GctBundle.getString("appengine.deployment.error.staging.docker.directory"));
```

Comment:
yea, in my UX update docs I change this so that the facet dialog can be saved even with errors.

I'd suggest to make the minimal set of changes here in this PR to accommodate updating to a Docker directory since once we agree on the UX updates there will be a separate PR to revamp this.